eval/eval_test.go

```diff
@@ -86,6 +87,33 @@ func Test_Eval(t *testing.T) {
 	}
 }
 
+func Test_Uuid(t *testing.T) {
+	expr := `Uuid = uuid; write _ $Uuid;`
+	nn, err := syntax.Parse("-", strings.NewReader(expr), errh(t))
+
+	if err != nil {
+		t.Fatalf("%s\n", err)
+	}
+
+	buf := &bytes.Buffer{}
+	e := New(buf)
+
+	err = e.Run(nn)
+
+	if err != nil {
+		t.Fatalf("expected evaluation of %q to be successful\n", expr)
+	}
+
+	uuidV4Format := `^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$`
+	matched, err := regexp.MatchString(uuidV4Format, buf.String())
+	if err != nil {
+		t.Fatalf("could not parse regex %s\n", uuidV4Format)
```

Comment:
Prefer `%q` over `%s` when providing error messages in test output, it makes it easier to delineate a value used in a test from every other part of the output message. Same for line 113 too.